src/measurement/vmsa.rs

```diff
@@ -113,6 +113,71 @@ where
     }
 }
 
+impl<T, const N: usize> TryFrom<Vec<T>> for LargeArray<T, N>
+where
+    T: std::marker::Copy + std::default::Default + for<'a> Deserialize<'a> + Serialize,
+{
+    type Error = LargeArrayError;
+
+    fn try_from(vec: Vec<T>) -> Result<Self, Self::Error> {
```

Comment:
If we go with the top thing...

```suggestion
    fn try_from(vec: &[T]) -> Result<Self, Self::Error> {
```